packages/frontend/src/app/paginated-list/paginated-list.component.spec.ts

```diff
@@ -12,123 +12,287 @@
-      expect(component.pageSize).toEqual(expectedPageSize);
-    });
-
-    it('should not update the rendered elements if no page was provided', () => {
-      component.pageIndex = 6;
-      component.pageSize = 3;
+      expect(component.viewConfig.pageIndex).toEqual(10);
+    }));
 
-      const expectedPageIndex = component.pageIndex;
-      const expectedPageSize = component.pageSize;
+    it('should set the previous page index when the previous page is selected', fakeAsync(() => {
+      spyOn(component, 'fetchPageData').and.returnValue(empty());
+      const page = {
+        pageIndex: 0,
+        previousPageIndex: 1,
+      } as PageEvent;
 
-      component.updatePage();
+      component.onPageSelectionChange(page);
+      tick();
 
-      expect(component.pageIndex).toEqual(expectedPageIndex);
-      expect(component.pageSize).toEqual(expectedPageSize);
-    });
+      expect(component.viewConfig.pageIndex).toEqual(0);
+    }));
 
-    it('should hide the paginator when there is only 1 page to show', () => {
-      // fill with the exact page size
-      component.showPaginator = true;
-      component.pageSize = 5;
-      component._elements = [];
-      for (let index = 0; index < component.pageSize; index++) {
-        component._elements.push({name: '', organization: ''});
-      }
+    it('should fetch the next page when the next page is selected', fakeAsync(() => {
+      const fetcher = spyOn(component, 'fetchPageData').and.returnValue(
+        empty()
+      );
+      const page = {
+        pageIndex: 1,
+        previousPageIndex: 0,
+      } as PageEvent;
 
-      component.updatePage();
+      component.onPageSelectionChange(page);
+      tick();
+
+      const expectedFilters: Filter[] = [
+        {name: 'limit', value: '10'},
+        {name: 'offset', value: '10'},
+      ];
+      expect(fetcher).toHaveBeenCalledTimes(1);
+      const passedArgs = fetcher.calls.mostRecent().args as object[];
+      expectedFilters.forEach(filter =>
+        expect(passedArgs[0]).toContain(jasmine.objectContaining(filter))
+      );
+    }));
+
+    it('should fetch the previous page when the previous page is selected', fakeAsync(() => {
+      const fetcher = spyOn(component, 'fetchPageData').and.returnValue(
+        empty()
+      );
+      const page = {
+        pageIndex: 0,
+        previousPageIndex: 1,
+      } as PageEvent;
 
-      expect(component.showPaginator).toBeFalse();
+      component.onPageSelectionChange(page);
+      tick();
+
+      const expectedFilters: Filter[] = [
+        {name: 'limit', value: '10'},
+        {name: 'offset', value: '0'},
+      ];
+      expect(fetcher).toHaveBeenCalledTimes(1);
+      const passedArgs = fetcher.calls.mostRecent().args as object[];
+      expectedFilters.forEach(filter =>
+        expect(passedArgs[0]).toContain(jasmine.objectContaining(filter))
+      );
+    }));
+
+    it('should restore the previous index if an error occurs while fetching data', fakeAsync(() => {
```

Comment:
This is important because the user should be allowed to select the next page again if something went wrong the first time.